pkg/controller/metrics/provision_underway_collector.go

```diff
@@ -217,5 +228,14 @@ func getKnownConditions(conditions []hivev1.ClusterDeploymentCondition) (conditi
 			}
 		}
 	}
-	return condition, reason
+	// if known condition not found, but some condition is in undesired state, report it
+	if condition == "Unknown" && len(conditions) > 0 {
+		// conditions would be sorted to have error conditions on the top, so if the first condition is in desired state,
+		// we can skip reporting it
+		if conditions[0].Status == corev1.ConditionUnknown ||
+			controllerutils.IsConditionInDesiredState(conditions[0]) {
+			skip = true
+		}
```

Comment:
- It's still ooking me out a bit how tightly bound this is to the sorting algorithm housed entirely elsewhere.
- If the top condition is not in the desired state, why don't we report *that*, rather than "Unknown"?